query/src/test/java/com/connexta/replication/query/QueryManagerTest.java

```diff
@@ -0,0 +1,121 @@
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package com.connexta.replication.query;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.connexta.replication.api.data.Site;
+import com.connexta.replication.api.data.SiteType;
+import com.connexta.replication.api.queue.SiteQueue;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class QueryManagerTest {
+
+  private static final long PERIOD = 60L;
+
+  private static final String SITE_ID = "siteid";
+
+  @Mock QueryServiceTools queryServiceTools;
+
+  @Mock ScheduledExecutorService executor;
+
+  @Mock private Site site;
+
+  @Mock private QueryService queryService;
+
+  private Stream<String> sites = Stream.empty();
+
+  private QueryManager queryManager;
+
+  @Before
+  public void setUp() throws Exception {
+    queryManager = new QueryManager(sites, queryServiceTools, executor);
+    when(queryServiceTools.getPeriod()).thenReturn(PERIOD);
+    when(site.getId()).thenReturn(SITE_ID);
+    when(site.getType()).thenReturn(SiteType.DDF);
+  }
+
+  @Test
+  public void init() {
+    queryManager.init();
+    verify(executor)
+        .scheduleAtFixedRate(
+            any(Runnable.class),
+            eq(TimeUnit.MINUTES.toSeconds(1)),
+            eq(PERIOD),
+            eq(TimeUnit.SECONDS));
+  }
+
+  @Test
+  public void destroy() {
+    queryManager.services.put(SITE_ID, queryService);
+    queryManager.destroy();
+    verify(executor).shutdownNow();
+    verify(queryService).stop();
+  }
+
+  @Test
+  public void reloadSiteConfigs() {
+    when(queryServiceTools.allSites()).thenReturn(Stream.of(site));
+    when(queryServiceTools.getQueueFor(site)).thenReturn(mock(SiteQueue.class));
+    queryManager.reloadSiteConfigs();
+    // assert that a query service was started
+    assertThat(queryManager.services.get(SITE_ID).future, Matchers.notNullValue());
```

Comment:
❗️ Avoid accessing private members of another class to verify something. Instead, expose a package private method to check its state that hides how to do so. Something like `isStarted()` maybe

Reply:
❓ Should we also verify that the service is being tracked internally now?

Reply:
tracked internally?

Reply:
Meaning that the manager actually is managing that service such that when it shuts down, it will shut it down for example?

Reply:
BTW, the manager needs to define a destroy() method so it can be registered with spring to be told to shutdown

Reply:
this is checking if the service was started by getting it from the managers list of services. I'm not sure how else we would check if they're tracked internally.